src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+                          5//12    -4//3    23//12    0
+                          -9//24   37//24   -59//24   55//24]
+
+##Adams Moulton Coefficients for Implicit Method
+const am_imp_coefficients3 =   [1         0        0         0
+                                1//2      1//2     0         0
+                                -1//12    8//12    5//12     0
+                                1//24     -5//24   19//24    9//24]
+
+
+const γ_star_array=[1.0, -0.5,-0.08333333333333334,-0.041666666666666664,
+                            -0.026388888888888885,-0.018750000000000003,
+                            -0.014269179894179893,-0.01136739417989418,
+                            -0.009356536596119928,-0.00789255401234568,
+                            -0.006785849984634706,-0.005924056412337663,
+                            -0.005236693257950285,-0.004677498407042264,
+                            -0.004214952239005473,-0.003826899553211884,
+                            -0.003497349845349918]
+"""
+#     t,y = ODE113(odefun, y0, tspan,order=4;kwords...)
+#
+#with tspan = [t0,tfinal] or[t0,t1,...,tfinal] for dense output
+#
+#Variable Step Variable Order (VSVO) Adams-Moulton Solver for nonstiff problems
+#
+#(Main ref: Hairer & Wanner 1996, Vol I, p.356-360, p.398-400, p.421-423)
+"""
+function ode113(F, y0, tspan; kwords...)
+    # For y0 which don't support indexing.
+    F_ = (t, y) -> [F(t, y[1])]
+    t,y = ode113(F_, [y0], tspan; kwords...)
+    return t, vcat_nosplat(y)
+end
+
+"a variable stepsize, variable order multi-step solver for nonstiff ODEs"
+function ode113(F, y0::AbstractVector, tspan; order::Integer = 4,
+                                        adaptive_order = true,
+                                        max_order = 13,
+                                        reltol = 1.0e-9,
+                                        abstol = reltol,
+                                        minstep = abs(tspan[1] - tspan[end])/1e18,
+                                        maxstep = abs(tspan[end] - tspan[1])/2.5,
+                                        initstep = 0.,
+                                        fac = .8,
+                                        norm = Base.norm,
+                                        points = :all,
+                                        progressmeter = false)
+
+    #-##########################################################################
+    # Initializations
+    #-##########################################################################
+    Et, Eyf, Ty = make_consistent_types(F, y0, tspan)
+    dof = length(y0)
+
+    # initialization of time span, which will have variable length
+    tspan = convert(Vector{Et}, tspan)
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = [tstart]
+
+    # initialzation of y[n], which will have variable length as well
+    # TODO: fix after https://github.com/JuliaLang/julia/issues/16667
+    #y_n = deepcopy(y0)
```

Comment:
To be fully compatible with `AbstractVector` this needs to deepcopy. You can however

```julia
if typeof(y0) <: Vector
  y_n = copy(y0)
else
  y_n = deepcopy(y0)
end
```

to be slightly more performant in the standard case that it's a standard vector.